File: Nth Fibonacci Number[13-August-2023]/Solution.java
// >> JAVA CODE <<

class Solution {
    static int nthFibonacci(int n){
        if(n==0) return 0;
        
        int a=0,b=1;
        int mod=(int)1e9+7;
        n-=2;
        while(n-->=0)
        {
            int c=(a%mod+b%mod)%mod;
            a=b;
            b=c;
        }
        
        return b;
    }
}

File: Nth Fibonacci Number[13-August-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    def nthFibonacci(self, n : int) -> int:
        if(n==0): return 0;
        a,b=0,1
        n-=2
        
        mod=1e9+7
        
        while(n>=0):
            c=(a%mod+b%mod)%mod
            a=b
            b=c
            
            n-=1
        
        return int(b%mod)